Fix bpf_sock compilation for ipv6-only

alexferenets opened the pull request:
Otherwise,  when started with `--enable-ipv4=false --enable-ipv6=true` I got
```
level=error msg="Failed to compile bpf_sock.o: exit status 1" compiler-pid=2393 subsys=datapath-loader
level=warning msg="/var/lib/cilium/bpf/bpf_sock.c:1094:10: error: implicit declaration of function '__sock4_health_fwd' [-Werror,-Wimplicit-function-declaration]" subsys=datapath-loader
level=warning msg="                return __sock4_health_fwd(ctx);" subsys=datapath-loader
level=warning msg="                       ^" subsys=datapath-loader
level=warning msg="/var/lib/cilium/bpf/bpf_sock.c:1094:10: note: did you mean '__sock6_health_fwd'?" subsys=datapath-loader
level=warning msg="/var/lib/cilium/bpf/bpf_sock.c:1086:1: note: '__sock6_health_fwd' declared here" subsys=datapath-loader
level=warning msg="__sock6_health_fwd(struct bpf_sock_addr *ctx __maybe_unused)" subsys=datapath-loader
level=warning msg=^ subsys=datapath-loader
level=warning msg="1 error generated." subsys=datapath-loader

```

borkmann commented on bpf/bpf_sock.c:
```diff
@@ -1089,8 +1089,10 @@ __sock6_health_fwd(struct bpf_sock_addr *ctx __maybe_unused)
 
 	ctx_get_v6_address(ctx, &addr6);
 	if (is_v4_in_v6(&addr6)) {
+#ifdef ENABLE_IPV4
```
I think we need to include also the `is_v4_in_v6()` so that if IPv4 is disabled we do all the lookups in the v6 maps.

YutaroHayakawa reviewed (approved):
We need to fix a point Daniel mentioned, but otherwise, looks good to me.

Commit:
Fix ipv6-only

Otherwise, when started with --enable-ipv4=false --enable-ipv6=true I got

level=error msg="Failed to compile bpf_sock.o: exit status 1" compiler-pid=2393 subsys=datapath-loader
level=warning msg="/var/lib/cilium/bpf/bpf_sock.c:1094:10: error: implicit declaration of function '__sock4_health_fwd' [-Werror,-Wimplicit-function-declaration]" subsys=datapath-loader
level=warning msg="                return __sock4_health_fwd(ctx);" subsys=datapath-loader
level=warning msg="                       ^" subsys=datapath-loader
level=warning msg="/var/lib/cilium/bpf/bpf_sock.c:1094:10: note: did you mean '__sock6_health_fwd'?" subsys=datapath-loader
level=warning msg="/var/lib/cilium/bpf/bpf_sock.c:1086:1: note: '__sock6_health_fwd' declared here" subsys=datapath-loader
level=warning msg="__sock6_health_fwd(struct bpf_sock_addr *ctx __maybe_unused)" subsys=datapath-loader
level=warning msg=^ subsys=datapath-loader
level=warning msg="1 error generated." subsys=datapath-loader

Signed-off-by: Alex Ferenets <al.ferenets@gmail.com>

borkmann commented on bpf/bpf_sock.c:
```diff
@@ -1105,7 +1107,9 @@ __sock6_health_fwd(struct bpf_sock_addr *ctx __maybe_unused)
 	return ret;
+#ifdef ENABLE_IPV4
 }
+#endif /* ENABLE_IPV4 */
```
What is this about?

borkmann commented on bpf/bpf_sock.c:
```diff
@@ -1088,9 +1088,11 @@ __sock6_health_fwd(struct bpf_sock_addr *ctx __maybe_unused)
 	if (is_v4_in_v6(&addr6)) {
 		return __sock4_health_fwd(ctx);
 	} else {
+#endif /* ENABLE_IPV4 */
```
Should the { go below here, so that when IPv4 is disabled the IPv6 code sits in the {} block?
